checkstyle-tester/diff.groovy

```diff
@@ -289,6 +292,38 @@ def generateSummaryIndexHtml(diffDir, checkstyleBaseReportInfo, checkstylePatchR
     println 'Creating report summary page finished...'
 }
 
+def generateConfigFileHtml(diffDir, configFile) {
+    def configFilenameNoExtension = getFilenameNoExtension(configFile.name)
+    def diffToolDir = Paths.get("").toAbsolutePath()
+        .parent
+        .resolve("patch-diff-report-tool")
+        .toFile()
+    def diffToolJarPath = getPathToDiffToolJar(diffToolDir)
+    this.class.classLoader.rootLoader.addURL(new URL("file:$diffToolJarPath"))
+
+    def textTransform = Class.forName("com.github.checkstyle.site.TextTransform").newInstance()
+    def configFileHtml = new File("$diffDir/${configFilenameNoExtension}.html")
+    textTransform.transform(configFile.name, configFileHtml.toPath().toString(), Locale.ENGLISH,
+        "UTF-8", "UTF-8")
+
+    return configFileHtml
+}
+
+def printConfigInfo(summaryIndexHtml, configFileHtml, configFilename) {
+    summaryIndexHtml << ('<h6>')
+    summaryIndexHtml << ("<a href='${configFileHtml.name}'>${configFilename} file</a>")
+    summaryIndexHtml << ('</h6>')
+}
+
+def getFilenameNoExtension(filename) {
```

Comment:
`getFilenameNoExtension` => `getFilenameWithoutExtension`.

I like to make code like reading a sentence. To me, `without` makes more grammatical sense.

Reply:
That's a great way to look at it, thanks for the advice.  Fixed.